src/main/java/com/aws/greengrass/componentmanager/plugins/Registry.java

```diff
@@ -51,6 +51,8 @@ public Registry(String endpoint) {
             this.type = ImageType.PRIVATE;
         } else {
             this.source = ImageSource.OTHER;
+            // Currently all registries other than ECR are assumed to be public since
+            // we do not have support for private registries other than ECR
```

Comment:
When we support private registries other than ECR, we will need to identify that based on if the component artifact also includes a reference to the source of credential, i.e. reference to the secret that holds its credential. It will be in the form of a new attribute like `secret:` along with the existing ones like `uri:`, so what this really needed is the entire artifact object instead of just the endpoint. I've refactored this a little bit to be able to plug more artifact attributes into this later